t/semaphore.t

```diff
@@ -0,0 +1,1068 @@
+            local sem = semaphore.new(0)
+            local ok, err = sem:wait(0.1)
+            ngx.say(tostring(ok).." "..tostring(err))
+        ';
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+false timeout
+--- no_error_log
+[error]
+
+
+
+=== TEST 3: basic semaphore wait not allow in body_filter_by_lua
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+        body_filter_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local sem = semaphore.new(0)
+            local ok, err = sem:wait(1)
+            ngx.log(ngx.ERR, err)
+        ';
+        return 200;
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+--- error_log
+API disabled in the context of (unknown)
+
+
+
+=== TEST 4: basic semaphore wait not allow in header_filter_by_lua
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+        header_filter_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local sem = semaphore.new(0)
+            local ok, err = sem:wait(1)
+            ngx.log(ngx.ERR, err)
+        ';
+        return 200;
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+--- error_log
+API disabled in the context of header_filter_by_lua*
+
+
+
+=== TEST 5: basic semaphore wait not allow in init_worker_by_lua
+--- http_config eval: $::HttpConfigIntWorkerBy
+--- config
+    location /test {
+       content_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local g = getmetatable(_G).__index
+            ngx.say(g.test)
+       ';
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+API disabled in the context of init_worker_by_lua*
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: basic semaphore wait not allow in set_by_lua
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+        set_by_lua $res '
+            local semaphore = require "ngx.semaphore"
+            local sem, err = semaphore.new(0)
+            local ok, err = sem:wait(1)
+            return err
+        ';
+        echo $res;
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+API disabled in the context of set_by_lua*
+--- no_error_log
+[error]
+
+
+
+=== TEST 7: basic semaphore wait not allow in log_by_lua
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+        log_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local sem = semaphore.new(0)
+            local ok, err = sem:wait(1)
+            ngx.log(ngx.ERR, err)
+            return err
+        ';
+        echo "try magics";
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+try magics
+--- error_log
+API disabled in the context of log_by_lua*
+
+
+
+=== TEST 8: basic semaphore new not allow in init_by_lua
+--- http_config eval: $::HttpConfigInitByLua
+--- config
+    location /test {
+        content_by_lua '
```

Comment:
This `content_by_lua` directive is useless for this test block and should be replaced with something trivial like `echo ok`. This is a horrible distraction for the reader.